src/utils.js

```diff
@@ -381,3 +381,23 @@ export const focusableElements = [
   'video[controls]',
   '[contenteditable]:not([contenteditable="false"])',
 ];
+
+export function addDefaultProps(defaultProps, props) {
+  if (!defaultProps || !props) return props;
+
+  let result = { ...props };
+
+  Object.keys(defaultProps).forEach((key) => {
+    if (result[key] === undefined) {
+      result[key] = defaultProps[key];
+    }
+    if (
+      Object.keys(defaultProps[key] || {}).length > 0 &&
+      typeof defaultProps[key] === 'object'
```

Comment:
This is not considering array (`typeof` array will be object, so it would be better to have something like `!Array.isArray(defaultProps[key])`)